clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4148,18 +4226,21 @@ void Sema::ConstructOpenCLKernel(FunctionDecl *KernelCallerFunc,
     // Attributes of a user-written SYCL kernel must be copied to the internally
     // generated alternative kernel, identified by a known string in its name.
     if (StableName.find("__pf_kernel_wrapper") != std::string::npos)
-      copySYCLKernelAttrs(KernelObj);
+      copySYCLKernelAttrs(KernelObj, KernelCallerFunc);
   }
 
-  bool IsSIMDKernel = isESIMDKernelType(KernelObj);
+  KernelCallOperatorVisitor KernelCallOperator(KernelCallerFunc, KernelObj);
```

Comment:
In this function one `KernelCallOperatorVisitor` can be re-used twice. And therefore only one `Traverse` will happen.
You can pass a reference to `KernelCallOperatorVisitor` to `copySYCLKernelAttrs` instead.